### PR TITLE
Discuss Euler EOC

### PR DESCRIPTION
**Pull request by MTCam**

### CODE REVIEW

File: mirgecom/euler.py [hunk 1]
```diff
@@ -115,29 +121,54 @@ def _aux_shape(ary, leading_shape):
...
+        aux_shape = single_valued([
+            _aux_shape(mass, ()),
+            _aux_shape(energy, ()),
+            _aux_shape(momentum, (dim,)),
+            _aux_shape(massfrac, (nspec,))])
+    else:
+        aux_shape = single_valued([
+            _aux_shape(mass, ()),
+            _aux_shape(energy, ()),
+            _aux_shape(momentum, (dim,))])
```
Comment (inducer) [inline mirgecom/euler.py:164]:
```suggestion
    aux_shapes = [
        _aux_shape(mass, ()),
        _aux_shape(energy, ()),
        _aux_shape(momentum, (dim,))])
    if massfrac is not None:
        nspec = len(massfrac)
        aux_shapes.append(_aux_shape(massfrac, (nspec,)))
    aux_shape = single_valued(aux_shapes)
```

File: mirgecom/euler.py [hunk 2]
```diff
@@ -150,17 +181,26 @@ def inviscid_flux(discr, eos, q):
...
     cv = split_conserved(dim, q)
     p = eos.pressure(cv)
 
     mom = cv.momentum
+
+    massfrac = None
+    if cv.massfrac is not None:
+        # mass fractions require a reshape here to get the right numpy
+        # broadcast behavior. Reshaped to [numspecies x 1] object.
+        massfrac = mom * cv.massfrac.reshape(-1, 1) / scalar(cv.mass)
```
Comment (inducer) [inline mirgecom/euler.py:197]:
```suggestion
        # massfrac shape: (numspecies, dim)
        massfrac = mom * cv.massfrac.reshape(-1, 1) / scalar(cv.mass)
```
(IMO knowledge of numpy broadcasting is assumed for someone reading this.)

Comment (inducer) [inline mirgecom/euler.py:197]: Is there a modification of the numerical flux/Riemann solver to match this?

**Comment by inducer** [comment]
Meshmode structured mesh generation lives [here](https://github.com/inducer/meshmode/blob/c0cb6d64ad69ade6a3c8aec2e8087148d38ae3c0/meshmode/mesh/generation.py#L753-L770) in case we'd like to exactly replicate the JSH/TW experiment.

**Comment by MTCam** [comment]
FYI, here's the wave-eager test_wave_accuracy test running an order-1 discretization, with a cubic test function:

```
Approximation error:
h      | Error              | Running EOC       
-------+--------------------+--------------------
0.25   | 24.894640451513872 |                   
0.125  | 13.637115001657211 | 0.8682967115351077
0.0625 | 7.142189810883077  | 0.9331000868344225

Overall EOC: 0.9006983991847657
```

**Comment by inducer** [comment]
If these are relative errors, then that's *way* pre-asymptotic.

**Comment by MTCam** [comment]
> If these are relative errors, then that's _way_ pre-asymptotic.

I pushed this as far as I could without running out of memory on Quartz.  The relative errors are still pretty high - but it seems to be going to 1.

**Comment by MTCam** [comment]
Here's an order 1 Euler RHS EOC test with trig function test field:

\rho = sin( (x - 1.0) / 2pi) + cos( (y - 1.0) / 2pi) 

```
----------------------------- Captured stdout call -----------------------------
EOC(1): h        | Error                  | Running EOC       
---------+------------------------+--------------------
0.25     | 0.003786717542714202   |                   
0.125    | 0.0019115264455846215  | 0.9862226526104727
0.0625   | 0.0009602659814729742  | 0.9932191852157362
0.03125  | 0.0004812537601042166  | 0.996636256657611 
0.015625 | 0.00024090644563107277 | 0.9983248182496275

Overall EOC: 0.9938661267340224
```
Sigh.

**Pull request closed without merging**